core/arch/x86_64/kernel/thread_private.h

```diff
@@ -120,14 +119,10 @@ void thread_excp_vect_workaround_a15(void);
 void thread_excp_vect_end(void);
 
 /*
- * Assembly function as the first function in a thread.  Handles a stdcall,
- * a0-a3 holds the parameters. Hands over to __thread_std_smc_entry() when
- * everything is set up and does some post processing once
- * __thread_std_smc_entry() returns.
+ * Handles a stdcall, r10-r15 holds the parameters
  */
-void thread_std_smc_entry(uint32_t a0, uint32_t a1, uint32_t a2, uint32_t a3);
-uint32_t __thread_std_smc_entry(uint32_t a0, uint32_t a1, uint32_t a2,
-				uint32_t a3);
+void thread_std_smc_entry(void);
```

Comment:
Why don't we adopt changes Arm code did? Like parameters change.

Reply:
yes, we can follow that